sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/stream/ForwardingClientResponseObserver.java

```diff
@@ -25,23 +25,27 @@
  * A {@link ClientResponseObserver} which delegates all {@link StreamObserver} calls.
  *
  * <p>Used to wrap existing {@link StreamObserver}s to be able to install an {@link
- * ClientCallStreamObserver#setOnReadyHandler(Runnable) onReadyHandler}.
+ * ClientCallStreamObserver#setOnReadyHandler(Runnable) onReadyHandler} and a handler invoked when
+ * the stream terminates.
  *
  * <p>This is as thread-safe as the underlying stream observer that is being wrapped.
  */
 public final class ForwardingClientResponseObserver<ReqT, RespT>
     implements ClientResponseObserver<RespT, ReqT> {
   public static <ReqT, RespT> ForwardingClientResponseObserver<ReqT, RespT> create(
-      StreamObserver<ReqT> inbound, Runnable onReadyHandler) {
-    return new ForwardingClientResponseObserver<>(inbound, onReadyHandler);
+      StreamObserver<ReqT> inbound, Runnable onReadyHandler, Runnable onDoneHandler) {
```

Comment:
public method signature changed. iiuc this is internal class, should it actually be package private; or keep the original overwrite method (without onDoneHandler)?

Reply:
It is internal AFAIK but it seems safer to add overload so went with that.